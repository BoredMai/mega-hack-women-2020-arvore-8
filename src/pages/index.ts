import { Page } from 'types';

import decorations from 'styles/decorations.module.css';

export const mountainPage: Page = {
  background: decorations.mountain,
  choice: {
    icon: 'mountain.png',
    label: 'Montanha',
  },
  content: [
    `Ahhh, a escolha das alturas!`,
    `Agora você deverá ir a montanha em forma de troll. Escale até a ponta do nariz dele. Entre na narina esquerda e suba até o olho.`,
    `Lá haverá uma lente de aumento que te mostrará onde a pequena fada brilhante estará. E assim ela o fez!`,
  ],
  foreground: decorations.princess,
};

export const forestPage: Page = {
  content: [],
  choice: {
    disabled: true,
    icon: 'forest.png',
    label: 'Floresta',
  },
};

export const destinationPage: Page = {
  background: decorations.library,
  content: [
    `Ele também pediu que {characterName} escolhesse com o coração.`,
    `Escolha o destino de {characterName}:`,
  ],
  foreground: decorations.princess,
  next: [mountainPage, forestPage],
};

export const bookPage: Page = {
  background: decorations.library,
  content: [
    `Ela pegou a escada, subiu até o último degrau, se esticou toda e conseguiu alcançá-lo.`,
    `- Ufa, finalmente – disse ela.`,
    `O livro repentinamente começou a falar e explicou que, após aberto, ela teria que completar uma tarefa.`,
  ],
  foreground: decorations.princess,
  next: [destinationPage],
};

export const libraryPage: Page = {
  background: decorations.library,
  content: [
    `A biblioteca do palácio era tão grande que, para percorrê-la por inteiro, levava-se três dias.`,
    `Um belo dia ela resolveu ir até o final, caminhou os três dias e foi até a última estante. Ao olhar para cima, viu um livro que brilhava e pensou "Será que é mágico?"`,
  ],
  foreground: decorations.princess,
  next: [bookPage],
};

export const wizardPage: Page = {
  background: decorations.library,
  choice: {
    icon: 'book.png',
    label: 'Estudiosa',
  },
  content: [
    `Princesa {characterName} amava estudar. Todos do povoado sempre sabiam onde encontrá-la. Quando alguém perguntava:`,
    `- Onde está {characterName}? Ela precisa almoçar!`,
    `O povoado inteiro respondia em peso:`,
    `- Princesa {characterName} está no lugar onde mora sua alegria, a biblioteca do palácio!`,
  ],
  foreground: decorations.princess,
  next: [libraryPage],
};

export const warriorPage: Page = {
  content: [],
  choice: {
    disabled: true,
    icon: 'sword.png',
    label: 'Guerreira',
  },
};

export const skillPage: Page = {
  background: decorations.castle,
  content: [`Princesa {characterName} era uma grande...`],
  foreground: decorations.princessWithBird,
  next: [warriorPage, wizardPage],
};

export const namePage: Page = {
  background: decorations.castle,
  content: [
    `O nome da princesa era...`,
    `Hmm... como era o nome dela mesmo?`,
    `{username}, escolha um nome para a princesa!`,
  ],
  foreground: decorations.princessWithBird,
  next: [skillPage],
  prompt: {
    key: 'characterName',
    type: 'text',
    placeholder: 'Nome da princesa',
  },
};

export const rootPage: Page = {
  background: decorations.castle,
  content: [
    `Era uma vez um reino misterioso e sagrado onde habitavam seres mágicos, um povoado e sua princesa.`,
    `Como ela era amada pelo seu povo!`,
    `Sua inteligência, coragem e lealdade para com seu povo era admirável.`,
  ],
  next: [namePage],
};
