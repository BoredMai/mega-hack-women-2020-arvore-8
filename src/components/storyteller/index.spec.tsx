import { shallow, ShallowWrapper } from 'enzyme';

import { rootPage, namePage } from 'pages';
import React from 'react';

import Storyteller from '.';

describe('Storyteller', () => {
  const setPageMock = jest.fn().mockName('setPage');
  const setStoryPromptMock = jest.fn().mockName('setStoryPrompt');
  const useStateSpy = jest.spyOn(React, 'useState');
  const username = 'Clarice';

  let wrapper: ShallowWrapper;

  describe('without branching', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useStateSpy
        .mockImplementationOnce(() => [rootPage, setPageMock])
        .mockImplementationOnce(() => [{}, setStoryPromptMock]);
      wrapper = shallow(<Storyteller username={username} />) as ShallowWrapper;
    });

    it('renders properly', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('with prompt', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useStateSpy
        .mockImplementationOnce(() => [namePage, setPageMock])
        .mockImplementationOnce(() => [{ username }, setStoryPromptMock]);
      wrapper = shallow(<Storyteller username={username} />) as ShallowWrapper;
    });

    it('renders properly', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('updates state correctly on promptUpdate', () => {
      const promptKey = 'promptKey';
      const promptValue = 'promptValue';
      wrapper.find('Prompt').simulate('promptUpdate', promptKey, promptValue);

      expect(setStoryPromptMock).toHaveBeenCalledWith({
        promptKey: promptValue,
        username,
      });
      expect(setPageMock).toHaveBeenCalledWith(namePage.next[0]);
    });
  });
});
