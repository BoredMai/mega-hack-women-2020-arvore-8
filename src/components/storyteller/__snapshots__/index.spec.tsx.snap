// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyteller renders properly 1`] = `
<div
  className="storyteller"
>
  <ContentWrapper>
    <p
      className="storyContent"
    >
      Era uma vez um reino misterioso e sagrado onde habitavam seres mágicos, um povoado e sua princesa! Como ela era amada pelo seu povo! Sua inteligência, coragem e lealdade para com seu povo era admirável.
Seu nome era...
    </p>
    <Prompt
      onPromptUpdate={[Function]}
      prompt={
        Object {
          "key": "characterName",
          "placeholder": "Nome da princesa",
          "type": "text",
        }
      }
    />
  </ContentWrapper>
</div>
`;
